batch/Dockerfile.test

```diff
@@ -7,4 +7,4 @@ RUN python3 -m pip install --no-cache-dir /hailtop \
 
 COPY batch/test/ /test/
 
-CMD ["python3", "-m", "pytest", "-v", "--instafail", "/test/"]
+CMD ["python3", "-m", "pytest", "-n", "2", "-v", "--instafail", "/test/"]
```

Comment:
`-n auto` autodetects number of cpus. Would it be less brittle to use that in all pytest invocations?

Reply:
I think [`auto`](https://github.com/pytest-dev/pytest-xdist/blob/9f9b70748115cf0da7ae06aa1fb1127b6ba757a9/src/xdist/plugin.py) is too brittle in the context of Docker. Consider, for example:

```
# docker run -it python:3.6 python -c 'import os; import multiprocessing; print(os.sched_getaffinity(0)); print(os.cpu_count()); print(multiprocessing.cpu_count());'
{0, 1, 2, 3}
4
4
# docker run --cpus 2 -it python:3.6 python -c 'import os; import multiprocessing; print(os.sched_getaffinity(0)); print(os.cpu_count()); print(multiprocessing.cpu_count());'
{0, 1, 2, 3}
4
4
# docker run --cpuset-cpus 0-1 -it python:3.6 python -c 'import os; import multiprocessing; print(os.sched_getaffinity(0)); print(os.cpu_count()); print(multiprocessing.cpu_count());'
{0, 1}
4
4
# docker run --cpu-quota 100000 -it python:3.6 python -c 'import os; import multiprocessing; print(os.sched_getaffinity(0)); print(os.cpu_count()); print(multiprocessing.cpu_count());'
{0, 1, 2, 3}
4
4
# docker run --cpu-period 100000 -it python:3.6 python -c 'import os; import multiprocessing; print(os.sched_getaffinity(0)); print(os.cpu_count()); print(multiprocessing.cpu_count());'
{0, 1, 2, 3}
4
4
```

Unless k8s is setting cpuset-cpus in docker (which I doubt it is, there's no reason to limit *which* cpu you use), it seems likely `pytest-xdist` will set `n` to 8 on our `n1-standard-8`s.